web/react/components/user_settings/user_settings_general.jsx

```diff
@@ -188,7 +205,8 @@ export default class UserSettingsGeneralTab extends React.Component {
         }
     }
     updateSection(section) {
-        this.setState(assign({}, this.setupInitialState(this.props), {clientError: '', serverError: '', emailError: ''}));
+        const emailChangeInProgress = this.state.emailChangeInProgress;
```

Comment:
Because it first calls setupInitialState which resets `emailChangeInProgress` to its default false, so to keep that from happening I added `emailChangeInProgress` to the list of stuff assign overwrites.  I assumed it was ok to do that since it we were already using assign but if not let me know and I can do it a different way.